tests/integration/conftest.py

```diff
@@ -123,3 +124,282 @@ def sample_video() -> str:
     path_to_video = 'tests/integration/media/cat.mp4'
     assert os.path.exists(path_to_video)
     return path_to_video
+
+
+@pytest.fixture
+def ontology():
+    bbox_tool = {
+        'required':
+            False,
+        'name':
+            'bbox',
+        'tool':
+            'rectangle',
+        'color':
+            '#a23030',
+        'classifications': [{
+            'required': False,
+            'instructions': 'nested',
+            'name': 'nested',
+            'type': 'radio',
+            'options': [{
+                'label': 'radio_option_1',
+                'value': 'radio_value_1'
+            }]
+        }]
+    }
+    polygon_tool = {
+        'required': False,
+        'name': 'polygon',
+        'tool': 'polygon',
+        'color': '#FF34FF',
+        'classifications': []
+    }
+    polyline_tool = {
+        'required': False,
+        'name': 'polyline',
+        'tool': 'line',
+        'color': '#FF4A46',
+        'classifications': []
+    }
+    point_tool = {
+        'required': False,
+        'name': 'point--',
+        'tool': 'point',
+        'color': '#008941',
+        'classifications': []
+    }
+    entity_tool = {
+        'required': False,
+        'name': 'entity--',
+        'tool': 'named-entity',
+        'color': '#006FA6',
+        'classifications': []
+    }
+    segmentation_tool = {
+        'required': False,
+        'name': 'segmentation--',
+        'tool': 'superpixel',
+        'color': '#A30059',
+        'classifications': []
+    }
+    checklist = {
+        'required':
+            False,
+        'instructions':
+            'checklist',
+        'name':
+            'checklist',
+        'type':
+            'checklist',
+        'options': [{
+            'label': 'option1',
+            'value': 'option1'
+        }, {
+            'label': 'option2',
+            'value': 'option2'
+        }, {
+            'label': 'optionN',
+            'value': 'optionn'
+        }]
+    }
+    free_form_text = {
+        'required': False,
+        'instructions': 'text',
+        'name': 'text',
+        'type': 'text',
+        'options': []
+    }
+
+    ####Why does this break?
+    #Adding radio buttons causes the whole ontology to be invalid..
+    #radio_buttons = {'required': False, 'instructions': 'radio_tool',  'name': 'radio_tool', 'type': 'radio', 'options': [{'label': 'yes', 'value': 'yes'}, {'label': 'no', 'value': 'no'}]},
+    #ontology = {"tools": [bbox_tool, polygon_tool, polyline_tool, point_tool, entity_tool, segmentation_tool], "classifications": [radio_buttons, checklist, free_form_text]}
+    ####
+    tools = [
+        bbox_tool, polygon_tool, polyline_tool, point_tool, entity_tool,
+        segmentation_tool
+    ]
+    classifications = [checklist, free_form_text]
+    return {"tools": tools, "classifications": classifications}
+
+
+@pytest.fixture
+def configured_project(client, rand_gen, ontology):
```

Comment:
1. I think `project_with_setup` is a better name.
2. I think this can reuse existing fixtures to be
```
@pytest.fixture
def project_with_setup(client, rand_gen, ontology, project, dataset):
    editor = list(client.get_labeling_frontends(
        where=LabelingFrontend.name == "editor"))[0]
    project.setup(editor, ontology)
    for _ in range(len(ontology['tools']) + len(ontology['classifications'])):
        dataset.create_data_row(row_data=IMG_URL)
    project.datasets.connect(dataset)
    yield project
```

Reply:
I'll reuse the dataset fixture but not the project one. A test could the project to not yet be setup.

Reply:
The default fixture scope is "function". This means the fixture is invoked (objects created and destroyed) for each test function. So if you reuse the `project` fixture in `project_with_setup`, that project will be created and destroyed for each test, without the setup contaminating between tests.

Reply:
You are right. I'll make the change